src/pages/signin/SignInPage.js

```diff
@@ -33,149 +30,157 @@ const propTypes = {
-        Performance.measureTTI();
+const SignInPage = ({account, credentials}) => {
+    const {translate, formatPhoneNumber} = useLocalize();
+    const {canUsePasswordlessLogins} = usePermissions();
+    const {isSmallScreenWidth} = useWindowDimensions();
 
+    useEffect(() => Performance.measureTTI(), []);
+    useEffect(() => {
         App.setLocale(Localize.getDevicePreferredLocale());
-    }
-
-    render() {
-        // Show the login form if
-        // - A login has not been entered yet
-        // - AND a validateCode has not been cached with sign in link
-        const showLoginForm = !this.props.credentials.login && !this.props.credentials.validateCode;
-
-        // Show the unlink form if
-        // - A login has been entered
-        // - AND the login is not the primary login
-        // - AND the login is not validated
-        const showUnlinkLoginForm =
-            this.props.credentials.login && this.props.account.primaryLogin && this.props.account.primaryLogin !== this.props.credentials.login && !this.props.account.validated;
-
-        // Show the old password form if
-        // - A login has been entered
-        // - AND an account exists and is validated for this login
-        // - AND a password hasn't been entered yet
-        // - AND haven't forgotten password
-        // - AND the login isn't an unvalidated secondary login
-        // - AND the user is NOT on the passwordless beta
-        const showPasswordForm =
-            Boolean(this.props.credentials.login) &&
-            this.props.account.validated &&
-            !this.props.credentials.password &&
-            !this.props.account.forgotPassword &&
-            !showUnlinkLoginForm &&
-            !Permissions.canUsePasswordlessLogins(this.props.betas);
-
-        // Show the new magic code / validate code form if
-        // - A login has been entered or a validateCode has been cached from sign in link
-        // - AND the login isn't an unvalidated secondary login
-        // - AND the user is on the 'passwordless' beta
-        const showValidateCodeForm = (this.props.credentials.login || this.props.credentials.validateCode) && !showUnlinkLoginForm && Permissions.canUsePasswordlessLogins(this.props.betas);
-
-        // Show the resend validation link form if
-        // - A login has been entered
-        // - AND is not validated or password is forgotten
-        // - AND the login isn't an unvalidated secondary login
-        // - AND user is not on 'passwordless' beta
-        const showResendValidationForm =
-            Boolean(this.props.credentials.login) &&
-            (!this.props.account.validated || this.props.account.forgotPassword) &&
-            !showUnlinkLoginForm &&
-            !Permissions.canUsePasswordlessLogins(this.props.betas);
-
-        let welcomeHeader = '';
-        let welcomeText = '';
-        if (showValidateCodeForm) {
-            if (this.props.account.requiresTwoFactorAuth) {
+    }, []);
```

Comment:
I would create just one `useEffect`, I think there is no need to have them in separates `useEffect`

Reply:
Again, this seems like a preference that we haven't standardized yet.

I could argue that one of the advantages of `useEffect` over the class component lifecycle is that it allows side-effects to be grouped more logically by what they're doing. i.e: setting the locale has nothing to do with benchmarking, so they should be in separate effects. That way if we add a dependency for the locale effect it has no effect on the performance benchmarking effect.